src/Symfony/Component/Mailer/Bridge/Sendgrid/Http/Api/SendgridTransport.php

```diff
@@ -113,9 +113,11 @@ private function getAttachments(Email $email): array
             $headers = $attachment->getPreparedHeaders();
             $filename = $headers->getHeaderParameter('Content-Disposition', 'filename');
             $disposition = $headers->getHeaderBody('Content-Disposition');
+            // Sendgrid does not accept line breaks for base64 encoded attachments
+            $content = 'base64' === $attachment->getEncoding() ? base64_encode($attachment->getBody()) : $attachment->bodyToString();
```

Comment:
this looks like bypassing the encoding abstraction
can't we instead decide for the encoding earlier? (a line-break-less variant of base64)

Reply:
We can have a specific base64 encoder, but it can’t be done earlier since the email is not aware of the transport. The transport would have to set the encoder on each attachment (DataPart instance) at send time